src/driver/linux_onload/ossock_calls.c

```diff
@@ -112,12 +114,18 @@ oo_fd_replace_file(struct file* old_filp, struct file* new_filp,
   }
 
   rcu_read_lock();
-  if( lookup_fdget_rcu(old_fd) != old_filp ) {
+  tmp_filp = lookup_fdget_rcu(old_fd);
+  if( tmp_filp != old_filp ) {
+    if( tmp_filp )
+      fput(tmp_filp);
     rcu_read_unlock();
```

Comment:
Why can't we use simple fget() here?

It is exactly what Ivan has asked.

Reply:
Yup, we can remove `lookup_fd_rcu()` and the kcompat for it entirely.

Reply:
Thanks guys! That's done.
\+ rebase on top of master